### ISSUE TITLE
Remove `SubResource` and `WritableSubResource` from resourcemanager

### ISSUE THREAD

**Issue by ArcturusZhang** [Azure/autorest.csharp#2216]
Since we have the safe flatten mechanism, and `SubResource` `WritableSubResource` only have one property, whenever the replacement happens, in most cases, it will be flattened right away.
From this perspective, there is no point to keep the `SubResource` and `WritableSubResource` models in the resourcemanager.
One exception is that we have a list of a thing that is replacable by `SubResource`, in this case, safe flatten will not take effect. In this case, we usually could have the concern of future changes which might introduce more properties to the replaced type and we would have a breaking change.

In summary, I suppose to replace models with only `Id` property with the `SubResource` and `WritableSubResource` is not a good idea.

**Comment by m-nash** [comment]
Wouldn't this be a breaking change?

**Comment by ArthurMa1978** [comment]
Might cause breaking change, revisit if it worth to do it now
